app/models/tasks/set_descriptors_handler.rb

```diff
@@ -4,10 +4,17 @@ module Tasks::SetDescriptorsHandler # rubocop:todo Style/Documentation
   # rubocop:todo Metrics/AbcSize
```

Comment:
Plan is to pretty much re-implement, rather than refactor this. Even ignoring the complexity of the code here, the Workflows controller is a bit fat. So this documentation is to try and help with identifying the key behaviours.

Reply:
Not all of which appear to be used or needed.